app/controllers/match_list_base_controller.rb

```diff
@@ -168,6 +168,16 @@ def available_programs
   end
   helper_method :available_programs
 
+  def available_contact_types
+    MatchContacts.input_names
+  end
+  helper_method :available_contact_types
+
+  def available_contacts
+    Contact.active_contacts
```

Comment:
Is this the right set? I thought about unioning the scopes for available contacts of each type, but that seems like potentially unnecessary complexity

Reply:
I believe so, though since you are only using it once, I'd probably not bother with the controller method and helper method, just shove `Contact.active_contacts` in the view.